Content.Server/Paper/PaperComponent.cs

```diff
@@ -1,111 +1,21 @@
         [DataField("content")]
         public string Content { get; set; } = "";
 
         [DataField("contentSize")]
         public int ContentSize { get; set; } = 500;
 
-
-        [ViewVariables] private BoundUserInterface? UserInterface => Owner.GetUIOrNull(PaperUiKey.Key);
-
-        protected override void Initialize()
-        {
-            base.Initialize();
-
-            if (UserInterface != null)
-            {
-                UserInterface.OnReceiveMessage += OnUiReceiveMessage;
-            }
-
-            Mode = PaperAction.Read;
-            UpdateUserInterface();
-        }
-
-        public void SetContent(string content)
-        {
-
-            Content = content + '\n';
-            UpdateUserInterface();
-
-            if (!_entMan.TryGetComponent(Owner, out AppearanceComponent? appearance))
-                return;
-
-            var status = string.IsNullOrWhiteSpace(content)
-                ? PaperStatus.Blank
-                : PaperStatus.Written;
-
-            appearance.SetData(PaperVisuals.Status, status);
-        }
-
-        public void UpdateUserInterface()
-        {
-            UserInterface?.SetState(new PaperBoundUserInterfaceState(Content, Mode));
-        }
-
-        public void Examine(FormattedMessage message, bool inDetailsRange)
-        {
-            if (!inDetailsRange)
-                return;
-            if (Content == "")
-                return;
-
-            message.AddMarkup(
-                Loc.GetString(
-                    "paper-component-examine-detail-has-words"
-                )
-            );
-        }
-
-        private void OnUiReceiveMessage(ServerBoundUserInterfaceMessage obj)
-        {
-            var msg = (PaperInputText) obj.Message;
-            if (string.IsNullOrEmpty(msg.Text))
-                return;
-
-
-            if (msg.Text.Length + Content.Length <= ContentSize)
-                Content += msg.Text + '\n';
-
-            if (_entMan.TryGetComponent(Owner, out AppearanceComponent? appearance))
-            {
-                appearance.SetData(PaperVisuals.Status, PaperStatus.Written);
-            }
-
-            _entMan.GetComponent<MetaDataComponent>(Owner).EntityDescription = "";
-            UpdateUserInterface();
-        }
-
-        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
-        {
-            if (!EntitySystem.Get<TagSystem>().HasTag(eventArgs.Using, "Write"))
-                return false;
-            if (!_entMan.TryGetComponent(eventArgs.User, out ActorComponent? actor))
-                return false;
-
-            Mode = PaperAction.Write;
-            UpdateUserInterface();
-            UserInterface?.Open(actor.PlayerSession);
-            return true;
-        }
+        [DataField("stampedBy")]
+        public List<string> StampedBy { get; set; } = new();
+        [ViewVariables] public BoundUserInterface? UserInterface => Owner.GetUIOrNull(PaperUiKey.Key);
```

Comment:
use the `UserInterfaceSystem` methods rather than `GetUIOrNull`, and don't put this in a proprety